Tests/Common/Storage/LocalObjectStoreTests.cs

```diff
@@ -305,6 +315,72 @@ public void DisposeRemovesEmptyStorageFolder()
             Assert.IsFalse(Directory.Exists("./LocalObjectStoreTests/unused"));
         }
 
+        [Test]
+        public void DisposeDoesNotDeleteStoreFiles()
+        {
+            string path;
+            using (var store = new LocalObjectStore())
+            {
+                store.Initialize("test", 0, 0, "", new Controls() {PersistenceIntervalSeconds = -1});
+                Assert.IsTrue(Directory.Exists("./LocalObjectStoreTests/test"));
+
+                var validData = new byte[1024 * 1024 * 4];
+                var saved = store.SaveBytes("a.txt", validData);
+                Assert.IsTrue(saved);
+
+                path = store.GetFilePath("a.txt");
+                Assert.IsTrue(File.Exists(path));
+            }
+
+            // Check that it still exists
+            Assert.IsTrue(File.Exists(path));
+        }
+
+        [Test]
+        public void QuantBookObjectStoreBehavior()
+        {
+            // Test for issue #4811, on loop store objects would duplicate
+            for (int i = 0; i < 3; i++)
+            {
+                // Create a QuantBook and save some data
+                var qb = new QuantBook();
+                qb.ObjectStore.Save("a.txt", "1010101010101010101010");
+                Assert.IsTrue(qb.ObjectStore.ContainsKey("a.txt"));
+
+                // Assert the store has only a.txt
+                var store = qb.ObjectStore.GetEnumerator().AsEnumerable().ToList();
+                Assert.IsTrue(store.Count == 1);
+                Assert.IsTrue(store[0].Key == "a.txt");
+
+                // Create a copy for use using GetFilePath, check that it exists
+                var path = qb.ObjectStore.GetFilePath("a.txt");
+                Assert.IsTrue(File.Exists(path));
```

Comment:
Nice tests! We should also add the 'Delete' case: We add a key that gets persisted and later we delete it. When we create a new object store it shouldn't re load it